src/redux/sagas/expenditures/cancelAndWaiveStreamingPayment.ts

```diff
@@ -0,0 +1,70 @@
+import { ClientType } from '@colony/colony-js';
+import { fork, put, takeEvery } from 'redux-saga/effects';
+
+import { type Action, ActionTypes, type AllActions } from '~redux/index.ts';
+
+import {
+  type ChannelDefinition,
+  createTransaction,
+  createTransactionChannels,
+  waitForTxResult,
+} from '../transactions/index.ts';
+import { initiateTransaction, putError, takeFrom } from '../utils/index.ts';
+
+function* cancelAndWaiveStreamingPaymentAction({
```

Comment:
It would be great to validate the user is the stream's recipient (contract requirements) before sending any transactions